sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -335,6 +336,32 @@ abstract class ExternalCatalog
     postToAll(RenameFunctionEvent(db, oldName, newName))
   }
 
+  final def checkSchemaContainsBucketingColumns(
```

Comment:
It will be very annoying to duplicate all checks in `SessionCatalog` and `ExternalCatalog`. I think we should put checks in `SessionCatalog` as it behaves like a front-end of `ExternalCatalog`